### PR TITLE
Hoopy's first draft of the install docs

### PR DESCRIPTION
**Pull request by matthewrevell**
This commit reworks the Nakama developer install docs, including:

 * Redrafted intro page.
 * Separated the types of install into different guides.
 * Recommend Docker-based install.

Work still to be done includes upgrades and instructions on specifying local storage for the containers.

### CODE REVIEW

File: docs/index.adoc [hunk 1]
```diff
@@ -1,48 +1,65 @@
...
 [docs__features-list]
 * image:/images/svg/built-for-the-cloud.svg[] link:./development/user.adoc[*User Accounts*]
-  Every player has a user account which can be authenticated by device IDs, social profiles, and many more, or by your own custom server.
+  Manage player accounts authenticated by device ID, social profiles or using your own custom server.
 * image:/images/svg/social-server.svg[] link:./development/user.adoc#_social_login[*Social Login*]
-  Add social login and enable all users to sign in or link with Facebook, Google, Steam, and other providers; and start playing right away.
+  Authenticate player logins using Facebook, Google, Steam and other providers.
 * image:/images/svg/data-backup.svg[] link:./development/storage.adoc[*Data Storage*]
-  Create records for each user and global ones shared for all users. This could be save games, maps, items, and other gameplay objects.
+  Store and retrieve data – such as game saves, maps, highscore board, inventory items and other gameplay objects – using Nakama as your distrbuted data store.
```
Comment (mofirouz) [inline docs/index.adoc]: "distrbuted" -> "distributed"

**Comment by matthewrevell** [comment]
I've fixed the typo, thanks Mo.

I've now also reworked the Docker upgrade path; let me know if you're happy with this.

I think this is ready to merge but please do give it a sanity check. @ukmadlz please check this too.

**Comment by novabyte** [comment]
@matthewrevell Nice 👍 I'll wait for @ukmadlz to review final edits and drop a note here before I merge it onto master.

**Comment by ukmadlz** [comment]
LGTM

**Comment by matthewrevell** [comment]
Mike and I discussed the upgrades for Docker and what I wrote isn't ideal. Let me push a revision before you merge.

**Comment by matthewrevell** [comment]
Okay, I've now added Mike's updates to the XXXs and fixed the incorrect Docker upgrade instructions. I'm happy for this to merge but we might go back and make minor changes to some of this section and then bundle them with other PRs.

**Comment by novabyte** [comment]
@matthewrevell Perfect. I'll merge it 👍

**Pull request merged**